File: python/semanticabi/metadata/EthTransferType.py
from enum import Enum


class EthTransferType(Enum):

    # primary transactions
    PRIMARY = 'Primary'
    # internal transactions
    INTERNAL = 'Internal'
    # block rewards
    REWARD = 'Reward'
    # erc20, erc721, erc115 transfers
    ERC = 'Erc'

File: js/src/semanticabi/metadata/EthTransferType.ts
import {Enum} from "common/Enum";

/**
 * @author zuyezheng
 */
export class EthTransferType extends Enum {

    static readonly PRIMARY = new this('primary');
    static readonly INTERNAL = new this('internal');
    static readonly REWARD = new this('reward');
    static readonly ERC = new this('erc');

}
EthTransferType.finalize();
